src/compiler/config/validate-config.ts

```diff
@@ -48,7 +61,7 @@ export const validateConfig = (userConfig?: Config) => {
   setBooleanConfig(config, 'sourceMap', null, typeof config.sourceMap === 'undefined' ? false : config.sourceMap);
   setBooleanConfig(config, 'watch', 'watch', false);
   setBooleanConfig(config, 'buildDocs', 'docs', !config.devMode);
-  setBooleanConfig(config, 'buildDist', 'esm', !config.devMode || config.buildEs5);
+  setBooleanConfig(config, 'buildDist', null, !config.devMode || config.buildEs5);
```

Comment:
I wasn't sure about that - this argument is supposed to be of type `keyof d.ConfigFlags` and `esm` is not a property on `ConfigFlags` so, assuming that `ConfigFlags` objects actually conform to that interface there shouldn't ever be a value under the `"esm"` key to pull out - I changed it for that reason, but it could be the case that we should keep it any, if we wanted to I think we'd need to add an `esm: boolean` prop to `ConfigFlags`